functions/pubsub/main_test.py

```diff
@@ -24,32 +24,29 @@
 FUNCTIONS_TOPIC = os.getenv("FUNCTIONS_TOPIC")
 
 
-@pytest.mark.skip("broken")
 def test_functions_pubsub_publish_should_fail_without_params():
     request = MagicMock()
-    request.body.topic = None
+    request.get_json.return_value = {}
     response = main.publish(request)
 
     assert 'Missing "topic" and/or "subscription" parameter.' in response
 
 
-@pytest.mark.skip("broken")
 def test_functions_pubsub_publish_should_publish_message():
     request = MagicMock()
-    request.body.topic = FUNCTIONS_TOPIC
-    request.body.message = "my_message"
+    request.get_json.return_value = {
+        "topic": FUNCTIONS_TOPIC, "message": "my_message"
+    }
 
     response = main.publish(request)
 
     assert response == "Message published."
 
 
-@pytest.mark.skip("broken")
 def test_functions_pubsub_subscribe_should_print_message(capsys):
-    pubsub_message = MagicMock()
-    pubsub_message.data = base64.b64encode(b"Hello, world!")
+    pubsub_message = {"data": base64.b64encode(b"Hello, world!") }
 
-    main.subscribe(pubsub_message)
+    main.subscribe(pubsub_message, None)
```

Comment:
I don't think we need an explicit `None`. If we really do, that should be fixed instead.

Reply:
It doesn't look like `context` is used in the function. How about just:

```python
def subscribe(event, context=None):
    # ...
```
?

Reply:
The function service is sending `context`, but our sample doesn't use it.
We decided to just pass None from our test.